### PR TITLE
Ignore publisher after cancelling subscription

### PR DESCRIPTION
**Pull request by zeal18**
The PR fixes a situation when a stream has been limited on the zio-streams side to a value less than the buffer size

### CODE REVIEW

File: src/main/scala/zio/interop/reactivestreams/Adapters.scala [hunk 1]
```diff
@@ -138,6 +138,9 @@ object Adapters {
       val subscriber =
         new Subscriber[A] {
 
+          def offerSafe(el: Exit[Option[Throwable], A]): UIO[Unit] =
+            ZIO.ifM(q.isShutdown)(ZIO.unit, q.offer(el).unit)
```
Comment (runtologist) [inline src/main/scala/zio/interop/reactivestreams/Adapters.scala:142]: How about replacing the `ifM` with an error handler? This way we do not have additional work on the happy path.

Scaladoc for `ZQueue.interrupt`: 
>  Interrupts any fibers that are suspended on `offer` or `take`. Future calls to `offer*` and `take*` will be interrupted immediately.

```suggestion
            q.offer(el).unit.catchAllCause(_ => UIO.unit)
```

File: src/test/scala/zio/interop/reactivestreams/PublisherToStreamSpec.scala [hunk 2]
```diff
@@ -130,6 +131,26 @@ object PublisherToStreamSpec extends DefaultRunnableSpec {
...
+      },
+      testM("ignores publisher calls after stream ending") {
+        withProbe(probe =>
+          assertM((for {
+            fiber  <- probe.toStream(bufferSize).runHead.fork
+            demand <- Task(probe.expectRequest())
+            _      <- Task(probe.sendNext(0))
+            _      <- Task(probe.expectCancelling())
+
+            _ <- TestClock.adjust(1.second)
```
Comment (runtologist) [inline src/test/scala/zio/interop/reactivestreams/PublisherToStreamSpec.scala:143]: Do we really need time here? AFAIKS the test works without.

**Review by runtologist — CHANGES_REQUESTED** [review]
Good catch @zeal18! Just two small comments.

File: src/test/scala/zio/interop/reactivestreams/PublisherToStreamSpec.scala (hunk 2 above)
Comment (zeal18) [reply src/test/scala/zio/interop/reactivestreams/PublisherToStreamSpec.scala]: oh, it's not needed, deleted it

File: src/main/scala/zio/interop/reactivestreams/Adapters.scala (hunk 1 above)
Comment (zeal18) [reply src/main/scala/zio/interop/reactivestreams/Adapters.scala]: it looks like `.catchAllCause(_ => UIO.unit)` ignores all errors, `unsafeRunSync` does the same with even less work)

**Comment by zeal18** [comment]
@runtologist thank you for the review, I've made some adjustments and pushed changes, please, take a look

File: src/main/scala/zio/interop/reactivestreams/Adapters.scala (hunk 1 above)
Comment (runtologist) [reply src/main/scala/zio/interop/reactivestreams/Adapters.scala]: That's even better. :+1:

**Review by runtologist — APPROVED** [review]

**Pull request merged**